Update Getting Started pages

milljm opened the pull request:
A new version of the package is available (fixes _weakref import).
Modify instructions where they were lacking, or otherwise wrong.

Closes #13639

Commit:
Update Getting Started pages

A new version of the package is available (fixes _weakref import).
Modify instructions where they were lacking, or otherwise wrong.

Closes #13639

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/358230/) on 0359921 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/13641/site)

This comment will be updated on new commits.

Commit:
I like the original grammatical statement

milljm commented:
As discovered by @loganharbour, High Sierra users must install Xcode, not just Xcode command line tools as we specify in our Getting Started instructions. Do no merge, until I have updated those instructions.

Commit:
Update and Beautify the Getting Started pages

Mac OS has changed recently with how Xcode/CLT is available for installation.
Mac OS older than Mojave must obtain Xcode from Apple's Developer site (App Store
no longer works).

Closes #13639

loganharbour commented on modules/doc/content/getting_started/installation/osx_other.md:
```diff
@@ -0,0 +1,32 @@
+
+The following, is required to be installed before you can begin using MOOSE.
+
+Due to the unfortunate difficulty Apple has placed on its users wishing to develop on platforms older than Mojave, it is strongly encouraged users upgrade to Mojave.
```
Ha.

dschwen replied:
Downloading the mac OS Catalina Beta to stay ahead of the curve!

milljm replied:
Wonder if they will ever remove the need to install the headers package

Commit:
Fix grammatical naming

aeslaughter commented on modules/doc/content/getting_started/installation/osx_other.md:
```diff
@@ -0,0 +1,32 @@
+
+The following, is required to be installed before you can begin using MOOSE.
+
+Due to the unfortunate difficulty Apple has placed on its users wishing to develop on platforms older than Mojave, it is strongly encouraged users upgrade to Mojave.
```
I think the following should be at the top of the page.

!alert info title=Consider updating to Mojave
Due to the unfortunate difficulty Apple has placed on users wishing to develop on platforms older than Mojave, it is strongly encouraged users upgrade to Mojave.

Commit:
Annotating the need to upgrade

Commit:
Forgot to close the alert

Commit:
Placing annotation at top of page

Commit:
Fix alert annotation...

cticenhour commented:
I noticed these OSX instructions are now saying that users are required to install Xcode after CLT, but I don't see how we actually need both.....I get the caveat that we need to source CLT from Apple manually if we *already have* Xcode, but that's the only instance I see where we'd explicitly have both on a new user system. Why are we now requiring both here?

loganharbour commented:
It's needed for lldb to work on systems older than Mojave.

cticenhour commented:
Sorry, I should have clarified - I was referring to the Mojave instructions.

cticenhour commented:
Nevermind.....Looks like I didn't go far enough in testing lldb on a clean install of Mojave. Xcode is definitely required 😄

milljm commented:
A user found an issue with moose_profile I want to fix.

Commit:
Fixed moose_profile issue

The exporting of MODULEPATH has been fixed.

Refs #13641

permcody commented:
Roger - we have other breakage I'm attempting to fix.


On Thu, Jul 11, 2019 at 10:21 AM Jason Miller <notifications@github.com>
wrote:

> @permcody <https://github.com/permcody> This is sorta important to get
> in. The current available packages on our website are broken.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/13641?email_source=notifications&email_token=AAXFOIGEOD6OKBAYWGINEZDP65MXLA5CNFSM4H3K72X2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODZXHJ2Q#issuecomment-510555370>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAXFOIB4OVFPLM2O4AAFL3DP65MXLANCNFSM4H3K72XQ>
> .
>
